### PR TITLE
Add support for Symfony Mailer

### PR DESCRIPTION
**Pull request by bobvandevijver**
This MR add support for the new Symfony Mailer component. SwiftMailer is still supported and used by default.

In order to use the Symfony Mailer, you will need configure it explicitly.

### CODE REVIEW

File: Configuration/ConfigurationInterface.php [hunk 1]
```diff
@@ -2,52 +2,52 @@
...
-  public function getReceiver();
+  public function getSystemVersion(): string;
 
   /**
-   * Retrieve user information from the token, and return it in a single string
-   *
-   * @param TokenInterface|null $token
-   *
-   * @return string
+   * SwiftMailer representation of the error sender
```
Comment (tobias-93) [inline Configuration/ConfigurationInterface.php]:
```suggestion
   * Representation of the error sender
```

File: Configuration/ConfigurationInterface.php [hunk 2]
```diff
@@ -2,52 +2,52 @@
...
    */
-  public function getUserInformation(TokenInterface $token = null);
+  public function getSender();
 
   /**
-   * Retrieve the system version
-   *
-   * @return mixed
+   * SwiftMailer representation of the error receiver
```
Comment (tobias-93) [inline Configuration/ConfigurationInterface.php]:
```suggestion
   * Representation of the error receiver
```

File: DependencyInjection/KickinExceptionHandlerExtension.php [hunk 3]
```diff
@@ -21,9 +24,65 @@ class KickinExceptionHandlerExtension extends Extension
...
+
+  /**
+   * Configures the Swift Mailer handler
+   *
+   * @param ContainerBuilder $container
+   */
+  private function configureSwiftMailer(ContainerBuilder $container)
+  {
+    if (!class_exists('Swift_Mailer')) {
+      throw new InvalidConfigurationException('You selected SwiftMailer als mail backend, but it is not installed. Try running `composer req symfony/swiftmailer-bundle` or switch to the Symfony mailer in the configuration.');
```
Comment (tobias-93) [inline DependencyInjection/KickinExceptionHandlerExtension.php]:
```suggestion
      throw new InvalidConfigurationException('You selected SwiftMailer as mail backend, but it is not installed. Try running `composer req symfony/swiftmailer-bundle` or switch to the Symfony mailer in the configuration.');
```

File: DependencyInjection/KickinExceptionHandlerExtension.php [hunk 4]
```diff
@@ -21,9 +24,65 @@ class KickinExceptionHandlerExtension extends Extension
...
+
+  /**
+   * Configures the Symfony Mailer handler
+   *
+   * @param ContainerBuilder $container
+   */
+  private function configureSymfonyMailer(ContainerBuilder $container)
+  {
+    if (!class_exists('Symfony\Component\Mailer\Mailer')) {
+      throw new InvalidConfigurationException('You selected the Symfony mailer als mail backend, but it is not installed. Try running `composer req symfony/mailer` or switch to SwiftMailer in the configuration.');
```
Comment (tobias-93) [inline DependencyInjection/KickinExceptionHandlerExtension.php]:
```suggestion
      throw new InvalidConfigurationException('You selected the Symfony mailer as mail backend, but it is not installed. Try running `composer req symfony/mailer` or switch to SwiftMailer in the configuration.');
```

File: composer.json [hunk 5]
```diff
@@ -16,17 +16,24 @@
...
-    "symfony/filesystem": "~3.0|~4.0",
-    "symfony/dependency-injection": "~3.0|~4.0",
-    "symfony/security": "~3.0|~4.0",
-    "symfony/config": "~3.0|~4.0",
-    "symfony/event-dispatcher": "~3.0|~4.0",
-    "symfony/http-foundation": "~3.0|~4.0",
-    "symfony/http-kernel": "~3.0|~4.0",
-    "symfony/twig-bundle": "~3.0|~4.0",
+    "php": ">=7.3",
+    "symfony/filesystem": "~4.4|~5.0",
```
Comment (tobias-93) [inline composer.json:20]: What makes you remove Symfony 3 support?

**Review by tobias-93 — CHANGES_REQUESTED** [review]
Some small changes, but overall great job!

File: composer.json (hunk 5 above)
Comment (bobvandevijver) [reply composer.json:20]: Multiple reasons for no longer supported Sf3:
- The Mailer is not supported
- The new events are not available

It eases the maintenance.

**Pull request merged**